tests/core/gui/test_validators.py

```diff
@@ -19,6 +19,7 @@ def test_device_label_validator(qtbot, core_config):
     le.show()
 
     qtbot.keyClicks(le, "abcd")
+    qtbot.wait_until(lambda: le.text() == "abcd")
```

Comment:
I think we shoul put the asserts in the wait_until callback instead of just waiting for the text to be present (to avoid having to think about when the validation kicks in)

Reply:
Having the callback return a boolean value is actually the equivalent of asserting this value.